src/acl_platform.cpp

```diff
@@ -624,27 +625,24 @@ static void l_initialize_offline_devices(int offline_mode) {
 
     // If the user specified an offline device, then load it.
     // Search the shipped board defs for the device:
-    for (i = 0;
-         i < ACL_MAX_DEVICE &&
-         2 * i < sizeof(acl_shipped_board_cfgs) / sizeof(const char *) + 1;
-         i++) {
-      int is_present = 0;
-      if (!shipped_board_def_valid[i])
+    for (unsigned board_def_i = 0; board_def_i < ACL_MAX_DEVICE;
+         board_def_i++) {
+      if (!shipped_board_def_valid[board_def_i])
         continue;
```

Comment:
I changed the size to the size of `shipped_board_def` instead since depending on the conditions of `if (offline_mode == ACL_CONTEXT_MPSIM)` , `shipped_board_def` may have `acl_shipped_board_cfgs.size()` or `acl_shipped_board_cfgs.size() +1` elements. This is due to the implementation of the vector.